### ISSUE TITLE
Improve init defaults for TypeScript projects

### ISSUE THREAD

**Issue by aluanhaddad** [jspm/jspm-cli#2110]
TLDR: Improve the getting started experience by

  1. setting "defaultExtenion": "ts" automatically for TypeScript projects, making the jspm-cli even more intuitive.

  2. setting pre-selected transpiler to `typescript` if the local package main ends with `.tsx`

One of the lovely features of jspm is its interactive `init` command. In most cases it sets you up with the right defaults out of the box with an ease I think few cli tools can rival. This includes [default] option selection based on previous entries.
For example, when initializing a new TypeScript project we get
```cmd
~/source/app> jspm init

Init mode (Quick, Standard, Custom) [Quick]: Quick
...
     Enter the main entry point of your package within the src\ folder.

     This allows System.import('app') to resolve to src\/app.js.

SystemJS.config local package main [app.js]:
```
When we enter `app.ts` instead of `app.js`, we are greeted by the pleasant surprise:
```cmd
SystemJS.config local package main [app.js]: app.ts
...
SystemJS.config transpiler (Babel, Traceur, TypeScript, None) [typescript]:
```
`typescript` is pre-selected as the default transpiler, hit enter are we're set.
We get the following automatically
```JavaScript
transpiler: "plugin-typescript",
  packages: {
    "app": {
      "main": "app.ts",
      "meta": {
        "*.ts": {
          "loader": "plugin-typescript"
        }
      }
    }
  }
```
Unfortunately, we will hit errors as soon as we `import` any other .ts files from `app.ts`.
```TypeScript
import {square} from './utils/math'; // src/utils/math.ts
console.log(square(3));
```
// http://localhost:3000/src/utils/math.js 404 (Not Found)
The solution is to add `"defaultExtension": "ts"` to the app package config above.
Note: in versions of TypeScript prior to 2.0.0, it was possible to write.
```TypeScript
import {square} from './utils/math.ts';
```
But this was extremely fragile, destroyed code portability, and was never officially supported by the language.
As of TypeScript 2.0.0 it is a compiler error.

The suggestion here is that we improve the getting started experience by adding `"defaultExtension": "ts"` to the app package automatically for TypeScript projects, making the jspm UX even more intuitive.

A related area of friction is that the following does not work as expected.
```cmd
~/source/app> jspm init

Init mode (Quick, Standard, Custom) [Quick]: Quick
...
SystemJS.config local package main [app.js]: app.tsx
...
SystemJS.config transpiler (Babel, Traceur, TypeScript, None) [babel]:
```
`babel` is pre-selected as the default transpiler, hit enter and we will need to fix the generated config or re-run the command.

The suggestion here is to set the pre-selected transpiler to `typescript` if the local package main ends with `.tsx`. Furthermore, if the above suggestion is adopted it would be natural to set `"defaultExtension": "tsx"`.

I would be happy to submit PRs for any of these changes. 

#2103
#https://github.com/frankwallis/plugin-typescript/issues/162

**Comment by guybedford** [comment]
Great points. A PR here would be very welcome.

**Comment by crtormen** [comment]
@guybedford I will question this here, cause don't think I should open an issue for this, but it needs some reference for those who have the same doubts. 
I'm getting pretty confused with the questions descriptions in jspm init, and with the new jspm.config structure. Would be great to make a documentation explaining better what means each step. My project is angular2. 

First:
>  Init mode (Quick, Standard, Custom)

 - what's the differences among them?

> Local package name (recommended, optional):

-  Well, now I know that is the name I call in my System.import, but I've had to change it manually.

> package.json directories.baseURL: .
> package.json configFiles folder [src/]: src/
> Use package.json configFiles.jspm:dev? [No]: No
> SystemJS.config browser baseURL (optional): /
> SystemJS.config Node local project path [src/src/]: src/src/

- I don't understand why is this duplicating src/src. And what is my Node local project path? Its' not clear for me.

Let me know if you prefer I open an issue with these questions.

Thank you.

**Comment by guybedford** [comment]
Released in 0.17.0-beta.30.
